src/device.rs

```diff
@@ -74,46 +85,25 @@ impl Device {
     }
 
     pub async fn flush(&mut self) -> Result<(), Box<dyn Error>> {
-        const MTU_SIZE: usize = 20;
-
-        while !self.tx_buffer.is_empty() {
-            let mut buf = Vec::with_capacity(MTU_SIZE);
-            for _ in 0..MTU_SIZE {
-                if let Some(byte) = self.tx_buffer.pop_front() {
-                    buf.push(byte);
-                } else {
-                    break;
-                }
+        let chunks = self.tx_buffer.chunks(20);
 
-                // this could be nicer i guess
-            }
+        let write_type = if chunks.len() > 1 {
+            WriteType::WithoutResponse
+        } else {
+            WriteType::WithResponse
+        };
```

Comment:
At least for my device (`GB02`), this WriteType does not seem to work. When I execute a `flush` operation, which is only one message, thus triggering `WithResponse`, I get this error:
```
Starting scan...
Connected to GB02
Discovering services...
Found 9 characteristics
Scan complete
thread 'main' panicked at C:\Users\micro\Documents\dev\catprint-rs\src\device.rs:100:18:
called `Result::unwrap()` on an `Err` value: Other("Error { code: HRESULT(0x80650003), message: \"Das Attribut kann nicht geschrieben werden.\" }")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
(Sorry for the German Windows error)

If I restore the previous way of always sending `WithoutResponse`, it works flawlessly.

Is sending in the `WithResponse` mode required for your printer? If yes, please add a flag similar to the `supports_compression` flag based on the device name.